src/runCompletion.ts

```diff
@@ -35,7 +50,57 @@ export default async function runCompletion(
 
   const result = await autocomplete(requestData, timeout);
 
-  return result;
+  if (result?.results.length || !retry?.cancellationToken) {
+    return result;
+  }
+
+  return handleRetries(requestData, retry);
+}
+
+function handleRetries(
```

Comment:
This function is the most interesting in this pr, it will try to re-fetch suggestions on predefined interval until either the result is not empty or timeout